proparse/src/main/java/org/prorefactor/proparse/antlr4/ProgressLexer.java

```diff
@@ -724,25 +806,33 @@ private void newMacroRef2(String theText, FilePos refPos) throws IOException {
     }
     // We must expand macros even if consuming,
     // because we can have &ENDIF inside a preprocesstoken
-    currentInput = new InputSource(++sourceCounter, new BufferedReader(new StringReader(theText)));
-    currentInclude.inputVector.add(currentInput);
     // For a macro/argument expansion, we use the file/line/col of
     // the opening curly '{' of the ref file, for all characters/tokens.
-    currentInput.enableMacroExpansion();
-    currentInput.fileIndex = refPos.file;
-    currentInput.nextLine = refPos.line;
-    currentInput.nextCol = refPos.col;
+    currentInput = new InputSource(++sourceCounter, theText, refPos.file, refPos.line, refPos.col);
+    currentInclude.addInputSource(currentInput);
+    currentInput.setNextLine(refPos.line);
+    currentInput.setNextCol(refPos.col);
   }
 
   /**
    * Cleanup work, once the parse is complete. Gets run by doParse even if there's an exception thrown.
    */
-  void parseComplete() throws IOException {
+  public void parseComplete() throws IOException {
     // Things to do once the parse is complete
     // Release input streams, so that files can be written to by other processes.
     // Otherwise, these hang around until the next parse or until the Progress
     // session closes, and nothing else can write to the include files.
+
+    // List all file indexes.
+    // TODO Probably a much cleaner way to write that
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)